klaas/scripts/apply_regression_model.py

```diff
@@ -1,62 +1,102 @@
 import numpy as np
 import click
-from IPython import embed
 from sklearn.externals import joblib
-
 import yaml
-from klaas import check_extension, read_data, write_data
-
-@click.command()
-@click.argument('configuration_path', type=click.Path(exists=True, dir_okay=False, file_okay=True) )
-@click.argument('data_path', type=click.Path(exists=True, dir_okay=False, file_okay=True) )
-@click.argument('model_path', type=click.Path(exists=True, dir_okay=False, file_okay=True) )
-@click.argument('predictions_path', type=click.Path(exists=False, dir_okay=False, file_okay=True) )
-def main(configuration_path, data_path, model_path, predictions_path):
-    '''
-    Apply loaded model to data. The cuts applied during model training will also be applied here.
-    WARNING: currently only taking 1 off position into account.
+import logging
+import h5py
+from tqdm import tqdm
 
-    CONFIGURATION_PATH: Path to the config yaml file.
+from ..io import read_h5py_chunked
+from ..preprocessing import convert_to_float32, check_valid_rows
 
-    DATA_PATH: path to the FACT data.
 
-    MODEL_PATH: Path to the pickled model.
+@click.command()
+@click.argument('configuration_path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('data_path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('model_path', type=click.Path(exists=True, dir_okay=False))
+@click.option('-k', '--key', help='HDF5 key for pandas or h5py hdf5', default='events')
+@click.option('-n', '--n-jobs', type=int, help='Number of cores to use')
+@click.option('-y', '--yes', help='Do not prompt for overwrites', is_flag=True)
+@click.option(
+    '-N', '--chunksize', type=int,
+    help='If given, only process the given number of events at once',
+)
+def main(configuration_path, data_path, model_path, key, chunksize, n_jobs, yes):
+    '''
+    Apply given model to data. Two columns are added to the file, energy_prediction
+    and energy_prediction_std
 
-    PREDICTIONS_PATH: Path to the data with added prediction columns.
+    CONFIGURATION_PATH: Path to the config yaml file
+    DATA_PATH: path to the FACT data in a h5py hdf5 file, e.g. erna_gather_fits output
+    MODEL_PATH: Path to the pickled model
     '''
-    check_extension(predictions_path)
+    logging.basicConfig(level=logging.INFO)
+    log = logging.getLogger()
 
     with open(configuration_path) as f:
         config = yaml.load(f)
 
     training_variables = config['training_variables']
-    query = config['query']
 
+    with h5py.File(data_path) as f:
```

Comment:
shouldn't this also be able to open pandas_hdf5 files? if I try to apply a trained model to a pandas_hdf5 it can naturally not find certain keys